### PR TITLE
[backend] Service config

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

**Comment by adrianschroeter** [comment]
On Mittwoch, 11. September 2024, 17:49:34 CEST Frank Schreiner wrote:
> Why introducing a new config file instead of BSConfig to store the credentials?

For one it belongs to the service and not obs code.

Also read permissions may should be different to the BSConfig.pm one.

-- 

Adrian Schroeter ***@***.***>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH, Frankenstraße 146, 90461 Nürnberg, Germany
(HRB 36809, AG Nürnberg) Geschäftsführer: Ivo Totev

**Comment by M0ses** [comment]
Am Donnerstag, dem 12.09.2024 um 00:24 -0700 schrieb Adrian Schröter:
> On Mittwoch, 11. September 2024, 17:49:34 CEST Frank Schreiner wrote:
> > Why introducing a new config file instead of BSConfig to store the
> > credentials?
> 
> For one it belongs to the service and not obs code.

Then we should consider to use (and share) `/etc/obs/services` by
default 

or (IMHO more secure):

share `/etc/obs/services/<service_name>/` if it exists on the host.

> 
> Also read permissions may should be different to the BSConfig.pm one.
> 

Yeah - but I don`t see why sharing a directory which needs to be
readable by obsservicerun (the user running the container) should be
more secure then having only one user (obsrun) who needs to be able to
read BSConfig and using credentials only via ENV vars.

Anyway, if we would like to consider security aspects also, from my 
POV its more than only file permissions.

**Review by M0ses — APPROVED** [review]
LGTM - Approved

File: src/backend/call-service-in-container [hunk 1]
```diff
@@ -173,6 +178,7 @@ else
 fi
 
 CONTAINER_VOLUMES="-v $OUTEROUTDIR:$INNEROUTDIR -v $OUTERSRCDIR:$INNERSRCDIR -v $OUTERHOMEDIR:$INNERHOMEDIR -v $MOUNTDIR$INNERSCRIPTDIR:$INNERSCRIPTDIR:ro"
+[ -n "$USE_SERVICE_CONFIG" ] && CONTAINER_VOLUMES="$CONTAINER_VOLUMES -v /etc/obs/services/$USE_SERVICE_CONFIG:/etc/obs/services/$USE_SERVICE_CONFIG:ro"
```
Comment (M0ses) [inline src/backend/call-service-in-container:181]: Reminder to myself:

<nitpicking>
if we need to mount "/etc/obs/services" in future, we easily can change

`USE_SERVICE_CONFIG="scm-bridge"` to `USE_SERVICE_CONFIG="/scm-bridge"`

and

`-v /etc/obs/services/$USE_SERVICE_CONFIG:/etc/obs/services/$USE_SERVICE_CONFIG:ro`

to

`-v /etc/obs/services$USE_SERVICE_CONFIG:/etc/obs/services$USE_SERVICE_CONFIG:ro`

and a service which needs _ALL_ directories in  `/etc/obs/services` simply sets

`USE_SERVICE_CONFIG="/"`

**Comment by M0ses** [comment]
**Errors in circleCi are false positives** as the tests are not related to call-service-in-container

**Pull request merged**